spec/support/contexts/db_restore.rb

```diff
@@ -0,0 +1,96 @@
+require "active_record"
+require 'more_core_extensions/core_ext/hash'
+
+# Some config settings in the helper class are class methods instead of
+# constants to allow them to be lazy loaded, instead of loaded right away,
+# allowing the PostgresRunner to boot up `postgres` and set it's values if
+# necessary.
+class RestoreHelper
+  SPEC_DIR      = File.expand_path(File.join("..", ".."), __dir__)
+  LIB_DIR       = File.expand_path(File.join("..", "lib"), SPEC_DIR)
+  PG_DUMPFILE   = File.join(SPEC_DIR, "util", "data", "pg_dump.gz")
+  PG_BACKUPFILE = File.join(SPEC_DIR, "util", "data", "pg_backup.tar.gz")
+
+  def self.pg_port
+    @pg_port ||= defined?(PostgresRunner) ? PostgresRunner.port : nil
+  end
+
+  def self.default_restore_dump_opts
+    @default_restore_dump_opts ||= {
+      :local_file => PG_DUMPFILE,
+      :username   => "root",
+      :password   => "smartvm",
+      :hostname   => "localhost",
+      :port       => pg_port
+    }.delete_blanks
+  end
+
+  def self.restore_backup_opts
+    { :local_file => PG_BACKUPFILE }
+  end
+end
+
+# Required that you define `db_name` yourself in your specs
+shared_context "Database Restore Validation Helpers" do
+  let(:active_record_connection_info) do
+    {
+      "adapter"  => "postgresql",
+      "encoding" => "utf8",
+      "host"     => "localhost",
```

Comment:
Again, I feel like connecting using the socket would simplify this. Or does that not work if multiple servers are running on the same system?

Reply:
Resolving this as it was answered in a different thread.